brayns/engine/systems/GenericBoundsSystem.h

```diff
@@ -20,31 +20,36 @@
 
 #pragma once
 
-#include <brayns/engine/model/Model.h>
-#include <brayns/engine/model/ModelComponents.h>
-#include <brayns/engine/volume/VolumeView.h>
+#include <brayns/engine/model/systemtypes/BoundsSystem.h>
 
 namespace brayns
 {
-/**
- * @brief Adds a renderable volume to the model
- */
-class VolumeRendererComponent final : public Component
+template<typename Type>
+class GenericBoundsSystem : public BoundsSystem
 {
 public:
-    template<typename T>
-    VolumeRendererComponent(T volumeData)
-        : _volume(std::move(volumeData))
-        , _volumeView(_volume)
+    Bounds compute(const Matrix4f &matrix, Components &components) override
     {
-    }
+        auto &component = components.get<Type>();
+        auto &elements = component.elements;
+
+        Bounds result;
+#pragma omp parallel
+        {
+            Bounds local;
 
-    virtual Bounds computeBounds(const Matrix4f &transform) const noexcept override;
-    virtual void onCreate() override;
-    bool commit() override;
+#pragma omp for
+            for (size_t i = 0; i < elements.size(); ++i)
+            {
+                auto &element = elements[i];
+                local.expand(element.computeBounds(matrix));
```

Comment:
I am not very familiar with OMP but here we have potentially multiple threads touching local no ?

thread1: for i in range(0, N) local.expand()
thread2: for i in range(N, M) local.expand()
...

Reply:
No, the parallelization block starts at `#pragma omp parallel` . `Bounds local;` is a local variable to each thread. Then we merge all together with a mutex section